python/ray/dag/collective_node.py

```diff
@@ -188,3 +198,7 @@ def _execute_impl(self, *args, **kwargs):
     @property
     def collective_op(self) -> _CollectiveOperation:
         return self._collective_op
+
+    @property
+    def synchronous_peers(self) -> List["CollectiveOutputNode"]:
```

Comment:
Comment for this API. Should we use the same name for the field?

Reply:
Removed `synchronous_peers`. Added `SynchronousGroup` interface.